aptos-move/aptos-vm-types/src/resolver.rs

```diff
@@ -19,6 +19,15 @@ use move_core_types::{language_storage::StructTag, value::MoveTypeLayout, vm_sta
 use move_vm_types::delayed_values::delayed_field_id::DelayedFieldID;
 use std::collections::{BTreeMap, HashMap};
 
+/// Gives a global (from an executing txn perspective) view into the synchronization aspects of the
+/// encompassing block execution. The goal is to improve the efficiency of overall block execution.
+/// For example, if the block execution has halted due to gas limit (committing only a prefix),
+/// ongoing speculative executions can return early. Similarly, if a read-write dependency is
+/// preficted between txns, it can allow the dependent txn to wait and avoid costly re-executions.
+pub trait BlockSynchronizationView {
```

Comment:
I think the PR over-engineers something that's really simple functionality. We just need to provide an abort (killswitch), so If going with a trait, call the trait KillSwichInterface (it's not really a view).

This BlockSynchornization name is very application specific, and doesn't really describe the desired functionality, but rather how it will be used.

Also `early_abort_execution` -> `was_signaled_for_abort` or something like that.

The comment above is more more confusing - like "Gives a global (from an executing txn perspective) view into the synchronization aspects of the encompassing block execution" seems like a lot of unnecessary info to just say provides an access to a boolean swich (or signal).

Reply:
Where is the over-engineering though, other than the name I don't see a shorter edit distance to getting functionality e2e?
If we do it this way however, I feel BlockSynchronizationView can have its merits: I highly doubt this will be the only similar functionality and we can have trait per desired method, or we could have a logical umbrella that provides the transaction a view into per-block context.
As for the name, I currently like @igor-aptos 's heartbeat most.

Reply:
Heartbeat sounds good, but I've seen `interruption/termination point` and `killswich` naming multiple times - and I think it better describes the intention (we are terminating something, and termination checks for that), whereas heartbeat is more like notifying others that we are still running. But I'd be OK with either.

My comment about overengineering was exactly about BlockSycnrhonizationView -> it only does one thing, so no reason to have a specific name, nor trying to encapsulate future functionality. If we end up with some set of traits that can be encapsulated as BlockSyncView, then we'll just define it as union of such traits in the future.

Having something that just a plain killswhich/heartbeat and implemented/documented as such is just much more straight forward.

Reply:
But yep, simple name/comment change would address my concerns (overengineering is in the comments/naming - attempting to be future proof)

Reply:
Killswitch defines the implementation we currently want.  But we might have a different functionality there later. 

Heartbeat defines VM responsibility in calling this function without opinion on its implementation. 

So I suggest calling the function heartbeat,  and having the implementation of the trait potentially be BlockSynchronizationKillSwitch or something specific like that

Reply:
If we want this natively supported in move VM, which is probably not a bad idea, I would make sure that the MoveVM implementation is agnostic to block STM, or other stuff. I would have a generic argument (which can be part of some other ones) that carries information on whether an interrupt was requested.

We can then define the granularity inside moveVM on which VM would check for a `requested_interrupt` and would act accordingly. I'd want that to be done generically so that in the case we don't want interrupts the compiler can compile out all the function calls.

I'd also have the API to be BlockSTM -> MoveVM calls `request_interrupt` or provides an implementation of it. MoveVM calls `interrupt_requesteed` at certain points (we can have multiple classes of points as well) and aborts with `INTERRUPTED` status.

Reply:
Also I like `BlockSynchronizationKillSwich`. but I don't think `heartbeat` defines what VM should be doing, it's more what BlockSTM is doing. VM `reacts` to a heartbeat, so `has_heartbeat` is might be a better name. But again, a more generic `interrupt_requested` is much more describing.